From: Alfred Chen <cchalpha@gmail.com>
Subject: [PATCH] bmq: [Sync] 14533a16c46d thermal/cpu-cooling, sched/core:
 Move the arch_set_thermal_pressure() API to generic scheduler code
Date: Wed, 3 Jun 2020 16:34:04 +0800

---
 kernel/sched/bmq.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/kernel/sched/bmq.c b/kernel/sched/bmq.c
index e4a8da074702d..5db8d6d50c493 100644
--- a/kernel/sched/bmq.c
+++ b/kernel/sched/bmq.c
@@ -2784,6 +2784,17 @@ unsigned long long task_sched_runtime(struct task_struct *p)
 	return ns;
 }
 
+DEFINE_PER_CPU(unsigned long, thermal_pressure);
+
+void arch_set_thermal_pressure(struct cpumask *cpus,
+			       unsigned long th_pressure)
+{
+	int cpu;
+
+	for_each_cpu(cpu, cpus)
+		WRITE_ONCE(per_cpu(thermal_pressure, cpu), th_pressure);
+}
+
 /* This manages tasks that have run out of timeslice during a scheduler_tick */
 static inline void scheduler_task_tick(struct rq *rq)
 {
